training/distributed_training/pytorch/data_parallel/maskrcnn/pytorch_smdataparallel_maskrcnn_demo.ipynb

```diff
@@ -153,8 +159,88 @@
     "\n",
     "1. You need use the same `subnet` and `vpc` and `security group` used with FSx when launching the SageMaker notebook instance. The same configurations will be used by your SageMaker training job.\n",
-    "2. Make sure you set appropriate inbound/output rules in the `security group`. Specically, opening up these ports is necessary for SageMaker to access the FSx filesystem in the training job. https://docs.aws.amazon.com/fsx/latest/LustreGuide/limit-access-security-groups.html\n",
-    "3. Make sure `SageMaker IAM Role` used to launch this SageMaker training job has access to `AmazonFSx`.\n"
+    "2. Make sure you set appropriate inbound/output rules in the `security group`. Specifically, opening up these ports is necessary for SageMaker to access the FSx file system in the training job. https://docs.aws.amazon.com/fsx/latest/LustreGuide/limit-access-security-groups.html\n",
+    "3. Make sure `SageMaker IAM Role` used to launch this SageMaker training job has access to `AmazonFSx`.\n",
+    "\n",
+    "You also can automatically create a FSx file system with the following command:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# use boto3 to create FSx\n",
+    "\n",
+    "fsx_response = fsx_client.create_file_system(\n",
+    "    FileSystemType=\"LUSTRE\",\n",
+    "    StorageCapacity=1200,\n",
+    "    StorageType=\"SSD\",\n",
+    "    SubnetIds=subnets,\n",
+    "    SecurityGroupIds=security_group_ids,\n",
+    "    Tags=[\n",
+    "        {\"Key\": \"Name\", \"Value\": \"COCO-storage\"},\n",
+    "    ],\n",
+    "    LustreConfiguration={\n",
+    "        \"WeeklyMaintenanceStartTime\": \"7:03:00\",\n",
+    "        \"ImportPath\": f\"s3://{bucket}/fsx_sync/\",  # where FSx will import data from in s3, can do entire bucket or a specific folder\n",
+    "        \"ImportedFileChunkSize\": 1024,\n",
+    "        \"DeploymentType\": \"PERSISTENT_1\",  # |'SCRATCH_1' |'SCRATCH_2' # PERSISTENT means the storage in FSx will be persistent, SCRATCH indicates the storage is temporary\n",
+    "        \"AutoImportPolicy\": \"NEW\",  # 'NONE'| |'NEW_CHANGED' # this policy is how often data will be imported to FSx from S3\n",
+    "        \"PerUnitStorageThroughput\": 200,  # this is specific to PERSISTENT storage, not required for temporary\n",
+    "    },\n",
+    ")\n",
+    "\n",
+    "fsx_response"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Download model\n",
+    "\n",
+    "If we are running our model in a VPC and restrict outside internet access, we'll need to download our base model ahead of time. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!wget https://dl.fbaipublicfiles.com/detectron/ImageNetPretrained/MSRA/R-50.pkl\n",
+    "!aws s3 cp R-50.pkl s3://{bucket}/pretrained_weights/R-50.pkl"
```

Comment:
I am thinking if it will be easier to add that to Dockerfile? 
wget https://dl.fbaipublicfiles.com/detectron/ImageNetPretrained/MSRA/R-50.pkl -P /root/.torch/models/

Then the new docker with maskrcnn deps will also have checkpoint in it, and each node will have the checkpoint ready to use inside its own docker, and you can remove the spot_ckpt as script will use it directly

Reply:
Hi sorry for the delay in responding. The only downside to that would be if they want to adjust the backbone they'll need to rebuild the docker image. I'll put the command in the dockerfile and then mention if they want to update it they can use spot_ckpt argument?